packages/example/ios/example_0_70_6Tests/RNTPrimerTests.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  RNTPrimerTests.swift
+//  example_0_70_6Tests
+//
+//  Created by Boris on 20.5.24..
+//
+
+import XCTest
+import PrimerSDK
+@testable import primer_io_react_native
+
+class RNTPrimerTests: XCTestCase {
+
+    var rnPrimer: RNTPrimer!
+
+    override func setUp() {
+        super.setUp()
+        rnPrimer = RNTPrimer()
+    }
+
+    override func tearDown() {
+        rnPrimer = nil
+        super.tearDown()
+    }
+
+    // MARK: - INITIALIZATION & REACT NATIVE SUPPORT
+
+    func testInitialization() {
+        XCTAssertNotNil(rnPrimer)
+        XCTAssertTrue(RNTPrimer.requiresMainQueueSetup())
+    }
+
+    func testSupportedEvents() {
+        let expectedEvents = PrimerEvents.allCases.map { $0.stringValue }
+        XCTAssertEqual(rnPrimer.supportedEvents(), expectedEvents)
+    }
+
+    // MARK: - SDK API
+
+    func testConfigure() {
+        let expectation = self.expectation(description: "Configure completes")
+
+        rnPrimer.configure("{\"setting\":\"value\"}", resolver: { _ in
+            expectation.fulfill()
+        }, rejecter: { _, _, _ in
+            XCTFail("Configure failed")
+        })
+
+        waitForExpectations(timeout: 2.0, handler: nil)
+    }
+
+    func testDismiss() {
+        rnPrimer.dismiss()
+        // No specific assertion as dismiss method doesn't return a value
+    }
+
+    func testCleanUp() {
+        rnPrimer.cleanUp()
+        // No specific assertion as cleanUp method doesn't return a value
+    }
+
+    // MARK: - DECISION HANDLERS
+
+    func testHandleTokenizationNewClientToken() {
```

Comment:
These tests are testing that the resolver is executed, but arent testing that the decision handler is called (the handler is always nil in these tests).

Probably for another ticket, but these could be expanded on.

In usage, the handler is set somewhere, then when it is executed on it is niled out.

So I guess we should test
- setting the handler
- assert it is called async
- assert it is nil afterwards

Reply:
Great point, and it would require a small refactoring of RNTPrimer by making vars internal instead of private, I will do it now.

Reply:
I was focused on writing tests without refactoring for now, but why not refactor? :) @NQuinn27

Reply:
Nice - if it is important that the resolver is called, maybe we need two expectations:
- one for the handler being executed (you added this in the refactor) ✅ 
- one for the resolver being called (we lost that in the refactor).

Reply:
Hmm I didnt notice it would require a change from `private` - `internal`. @jnewc please sanity check this and let me know if you think I am barking up the wrong tree...

Reply:
2 expectations were added for the handler and the resolver! @NQuinn27